pkg/controller/servicebinding/servicebinder.go

```diff
@@ -326,13 +325,33 @@ func (b *serviceBinder) bind() (reconcile.Result, error) {
 	})
 
 	if isApplicationEmpty(b.sbr.Spec.Application) {
-		return b.handleApplicationError(EmptyApplicationReason, errEmptyApplication, sbrStatus)
+		conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+			Type:    InjectionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  EmptyApplicationReason,
+			Message: errEmptyApplication.Error(),
+		})
+		conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+			Type:   BindingReady,
+			Status: corev1.ConditionTrue,
+		})
+		return b.handleApplicationError(errEmptyApplication, sbrStatus)
 	}
 	updatedObjects, err := b.binder.bind()
 	if err != nil {
 		b.logger.Error(err, "On binding application.")
 		if errors.Is(err, errApplicationNotFound) {
-			return b.handleApplicationError(ApplicationNotFoundReason, errApplicationNotFound, sbrStatus)
+			conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+				Type:    InjectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  ApplicationNotFoundReason,
+				Message: errApplicationNotFound.Error(),
+			})
+			conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+				Type:   BindingReady,
+				Status: corev1.ConditionFalse,
+			})
+			return b.handleApplicationError(errApplicationNotFound, sbrStatus)
```

Comment:
this was part of `handleApplicationError` - why taking it out from the function and placing here?

Reply:
@pedjak 
https://github.com/redhat-developer/service-binding-operator/pull/681#discussion_r496023405